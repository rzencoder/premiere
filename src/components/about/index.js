import React from "react";
import { Container, Item, Divider, Text, Title, Icon } from "./styles/about";

export default function About({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

About.Item = function AboutItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

About.Divider = function AboutDivider({ children, ...restProps }) {
  return <Divider {...restProps} />;
};

About.Title = function AboutTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

About.Text = function AboutText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

About.Icon = function AboutIcon({ src, ...restProps }) {
  return <Icon {...restProps} src={src} />;
};
