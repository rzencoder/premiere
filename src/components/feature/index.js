import React from "react";
import {
  Background,
  Container,
  Overlay,
  TextContainer,
  Title,
  Text,
  Button,
  Details,
  SubText,
  Rating,
  AgeRating,
  Year,
} from "./styles/feature";

export default function Feature({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Feature.Background = function FeatureBackground({ ...restProps }) {
  return <Background {...restProps} />;
};

Feature.Overlay = function FeatureOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Feature.TextContainer = function FeatureTextContainer({
  children,
  ...restProps
}) {
  return <TextContainer {...restProps}>{children}</TextContainer>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Text = function FeatureText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Feature.Button = function FeatureButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Feature.Details = function FeatureDetails({ children, ...restProps }) {
  return <Details {...restProps}>{children}</Details>;
};

Feature.SubText = function FeatureSubText({ children, ...restProps }) {
  return <SubText {...restProps}>{children}</SubText>;
};

Feature.AgeRating = function FeatureAgeRating({ ...restProps }) {
  return <AgeRating {...restProps} />;
};

Feature.Year = function FeatureYear({ children, ...restProps }) {
  return (
    <Year {...restProps}>
      <div>{children}</div>
    </Year>
  );
};

Feature.Rating = function FeatureRating({ children, ...restProps }) {
  return (
    <Rating {...restProps}>
      <div>{children}</div>
    </Rating>
  );
};
