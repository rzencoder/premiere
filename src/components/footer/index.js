import React from "react";
import {
  Container,
  Text,
  Subtitle,
  Section,
  Link,
  Divider,
  Column,
  Break,
} from "./styles/footer";

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Footer.Subtitle = function FooterSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Divider = function FooterDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Footer.Section = function FooterSection({ children, ...restProps }) {
  return <Section {...restProps}>{children}</Section>;
};

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Break = function FooterBreak({ ...restProps }) {
  return <Break {...restProps} />;
};
