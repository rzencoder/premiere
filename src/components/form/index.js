import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Container,
  Title,
  Content,
  Label,
  Input,
  Submit,
  Frame,
  Text,
  Link,
  Error,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Frame = function FormFrame({ children, ...restProps }) {
  return (
    <Frame {...restProps}>
      <div>{children}</div>
    </Frame>
  );
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Content = function FormContent({ children, ...restProps }) {
  return <Content {...restProps}>{children}</Content>;
};

Form.Label = function FormLabel({ children, ...restProps }) {
  return <Label {...restProps}>{children}</Label>;
};

Form.Input = function FormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.Link = function FormLink({ to, children, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Link {...restProps}>{children}</Link>
    </ReactRouterLink>
  );
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};
