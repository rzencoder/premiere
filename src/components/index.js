export { default as Navigation } from "./navigation";
export { default as Jumbotron } from "./jumbotron";
export { default as About } from "./about";
export { default as Footer } from "./footer";
export { default as Form } from "./form";
