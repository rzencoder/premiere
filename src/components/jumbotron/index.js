import React from "react";
import {
  Background,
  Container,
  Overlay,
  Group,
  Title,
  Text,
  CallToAction,
} from "./styles/jumbotron";
import { Link as ReactRouterLink } from "react-router-dom";

export default function Jumbotron({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Jumbotron.Overlay = function JumbotronOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Jumbotron.Group = function JumbotronGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.Text = function JumbotronText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Jumbotron.CallToAction = function JumbotronCallToAction({
  to,
  children,
  ...restProps
}) {
  return (
    <ReactRouterLink to={to}>
      <CallToAction {...restProps}>{children}</CallToAction>
    </ReactRouterLink>
  );
};

Jumbotron.Background = function JumbotronBackground({ ...restProps }) {
  return <Background {...restProps} />;
};
