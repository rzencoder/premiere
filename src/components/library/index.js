import React from "react";
import {
  Container,
  Image,
  Button,
  Divider,
  CarouselContainer,
  CarouselGroup,
  CarouselButton,
  Selector,
  SelectorItem,
  CarouselOverlay,
  Panel,
} from "./styles/library";

export default function Library({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Library.Button = function LibraryButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Library.Divider = function LibraryDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Library.Image = function LibraryImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Library.CarouselContainer = function LibraryCarouselContainer({
  children,
  ...restProps
}) {
  return <CarouselContainer {...restProps}>{children}</CarouselContainer>;
};

Library.CarouselGroup = function LibraryCarouselGroup({
  children,
  ...restProps
}) {
  return <CarouselGroup {...restProps}>{children}</CarouselGroup>;
};

Library.CarouselButton = function LibraryCarouselButton({
  children,
  ...restProps
}) {
  return <CarouselButton {...restProps}>{children}</CarouselButton>;
};

Library.Selector = function LibrarySelector({ children, ...restProps }) {
  return (
    <Selector {...restProps}>
      <div>{children}</div>
    </Selector>
  );
};

Library.SelectorItem = function LibrarySelectorItem({
  children,
  ...restProps
}) {
  return <SelectorItem {...restProps}>{children}</SelectorItem>;
};

Library.CarouselOverlay = function LibraryCarouselOverlay({ ...restProps }) {
  return <CarouselOverlay {...restProps} />;
};

Library.Panel = function LibraryPanel({ children, ...restProps }) {
  return (
    <Panel {...restProps}>
      <div>{children}</div>
    </Panel>
  );
};
