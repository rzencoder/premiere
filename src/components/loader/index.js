import React from "react";
import { Container, Spinner } from "./styles/loader";

export default function Loader({ ...restProps }) {
  return (
    <Container data-testid="loader">
      <Spinner {...restProps} />;
    </Container>
  );
}
