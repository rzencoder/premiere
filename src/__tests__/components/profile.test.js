import React from "react";
import { render } from "@testing-library/react";
import { Profile } from "../../components";

const user = {
  displayName: "Tester",
  photoURL: "2",
};

describe("<Profile />", () => {
  it("renders <Profile /> component", () => {
    const openProfile = true;
    const { container, getByText } = render(
      <Profile open={openProfile}>
        <Profile.Image
          tabIndex={0}
          user={user.photoURL}
          onClick={() => {}}
          onKeyDown={() => {}}
        >
          <Profile.ImageText>{user.displayName[0]}</Profile.ImageText>
        </Profile.Image>
        <Profile.Menu open={openProfile}>
          <Profile.Title>{`${user.displayName}'s Account`}</Profile.Title>
          <Profile.Link>Billing</Profile.Link>
          <Profile.Link>Settings</Profile.Link>
          <Profile.Link>Help</Profile.Link>
          <Profile.Link onClick={() => {}}>Sign Out</Profile.Link>
        </Profile.Menu>
      </Profile>
    );

    expect(getByText("Tester's Account")).toBeTruthy();
    expect(getByText("Billing")).toBeTruthy();
    expect(getByText("Sign Out")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders different style when profile closed", () => {
    const openProfile = false;
    const { container } = render(
      <Profile open={openProfile}>
        <Profile.Image
          tabIndex={0}
          user={null}
          onClick={() => {}}
          onKeyDown={() => {}}
        >
          <Profile.ImageText>{user.displayName[0]}</Profile.ImageText>
        </Profile.Image>
        <Profile.Menu open={openProfile}>
          <Profile.Title>{`${user.displayName}'s Account`}</Profile.Title>
          <Profile.Link>Billing</Profile.Link>
          <Profile.Link>Settings</Profile.Link>
          <Profile.Link>Help</Profile.Link>
          <Profile.Link onClick={() => {}}>Sign Out</Profile.Link>
        </Profile.Menu>
      </Profile>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
