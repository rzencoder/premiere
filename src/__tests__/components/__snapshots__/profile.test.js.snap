// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders <Profile /> component 1`] = `
<div
  class="profile__Container-sc-1noxl8m-0 fJsANY"
  open=""
>
  <div
    class="profile__Image-sc-1noxl8m-1 jWcOvY"
    tabindex="0"
  >
    <div
      class="profile__ImageText-sc-1noxl8m-2 gcoGKg"
    >
      T
    </div>
  </div>
  <div
    class="profile__Menu-sc-1noxl8m-3 iGBBEW"
    data-testid="profile-menu"
    open=""
  >
    <div
      class="profile__Title-sc-1noxl8m-4 gUDTQq"
    >
      Tester's Account
    </div>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Billing
    </button>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Settings
    </button>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Help
    </button>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Sign Out
    </button>
  </div>
</div>
`;

exports[`<Profile /> renders different style when profile closed 1`] = `
<div
  class="profile__Container-sc-1noxl8m-0 fJsANY"
>
  <div
    class="profile__Image-sc-1noxl8m-1 ldDBkO"
    tabindex="0"
  >
    <div
      class="profile__ImageText-sc-1noxl8m-2 gcoGKg"
    >
      T
    </div>
  </div>
  <div
    class="profile__Menu-sc-1noxl8m-3 gzUzXn"
    data-testid="profile-menu"
  >
    <div
      class="profile__Title-sc-1noxl8m-4 gUDTQq"
    >
      Tester's Account
    </div>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Billing
    </button>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Settings
    </button>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Help
    </button>
    <button
      class="profile__Link-sc-1noxl8m-5 dcRoGy"
    >
      Sign Out
    </button>
  </div>
</div>
`;
